src/frequenz/sdk/actor/_power_managing/_base_classes.py

```diff
@@ -0,0 +1,189 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Base classes for the power manager."""
+
+from __future__ import annotations
+
+import abc
+import dataclasses
+import datetime
+import enum
+import typing
+
+from ... import timeseries
+from ...timeseries import Power
+
+if typing.TYPE_CHECKING:
+    from ...timeseries.battery_pool import PowerMetrics
+    from .. import power_distributing
+
+
+@dataclasses.dataclass(frozen=True)
+class ReportRequest:
+    """A request to start a reporting stream."""
+
+    source_id: str
+    """The source ID of the actor sending the request."""
+
+    battery_ids: frozenset[int]
+    """The battery IDs to report on."""
+
+    priority: int
+    """The priority of the actor ."""
+
+    def get_channel_name(self) -> str:
+        """Get the channel name for the report request.
+
+        Returns:
+            The channel name to use to identify the corresponding report channel
+                from the channel registry.
+        """
+        return f"power_manager.report.{self.battery_ids=}.{self.priority=}"
+
+
+@dataclasses.dataclass(frozen=True)
+class Report:
+    """Current PowerManager report for a set of batteries."""
+
+    target_power: Power | None
+    """The currently set power for the batteries."""
+
+    inclusion_bounds: timeseries.Bounds[Power] | None
+    """The available inclusion bounds for the batteries, for the actor's priority.
+
+    These bounds are adjusted to any restrictions placed by actors with higher
+    priorities.
+    """
+
+    exclusion_bounds: timeseries.Bounds[Power] | None
```

Comment:
To make bounds keyword-only (**and we need to move them to the end**):
```suggestion
    inclusion_bounds: timeseries.Bounds[Power] | None = dataclasses.field(kw_only=True)
    """The available inclusion bounds for the batteries, for the actor's priority.

    These bounds are adjusted to any restrictions placed by actors with higher
    priorities.
    """

    exclusion_bounds: timeseries.Bounds[Power] | None = dataclasses.field(kw_only=True)
```

Reply:
Done here: https://github.com/frequenz-floss/frequenz-sdk-python/pull/692/commits/6dfde504d93558de8e421c9856c0fdc42ab6d1d7